### ISSUE TITLE
Incorrect Email Notification for Payment

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#54294]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.76-12
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @jamesdeanexpensify 
**Slack conversation** (hyperlinked to channel name): [expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1734466277835559)

## Action Performed:

1. Open a direct message (DM) chat.
2. Use the grey "+" button to send a payment (e.g., $1) to another user
3. Observe the email notification sent to the sender instead of the recipient.

## Expected Result:
The email notification regarding the payment should be sent to the recipient

## Actual Result:
The sender received the email notification intended for the recipient.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

![image (14)](https://github.com/user-attachments/assets/a6972913-5d63-4f2b-b627-3e34e4d6d5a1)

![Snip - Nata and Rajan - mnata utest@gmail com - Gmail - Google Chrome](https://github.com/user-attachments/assets/d174d70e-be24-46de-849d-ffad2df740b8)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sakluger** [comment]
I just got this email, which seems correct:

<img width="1073" alt="image" src="https://github.com/user-attachments/assets/69cd9101-0e93-40d1-8e8f-27a61dcbda91" />

**Comment by sakluger** [comment]
I agree that we shouldn't send that second email to the payment sender (`Please pay my $1.00 expense`), but I'm not sure if we want to send the correct email to both the sender & recipient or just the recipient. I asked for clarification on expected behavior in the original thread.

I also asked in `#engineering` whether this is internal or external: https://expensify.slack.com/archives/C03SSAQ7P/p1734563297141249

**Comment by sakluger** [comment]
I got confirmation that this is an internal issue.

**Comment by sakluger** [comment]
Adding to the expense project.

**Comment by sakluger** [comment]
Still no updates.

**Comment by sakluger** [comment]
I'll check on priority this week.

**Comment by sakluger** [comment]
Asked for a volunteer here: https://expensify.slack.com/archives/C03SSAQ7P/p1738168052703189

**Comment by Beamanator** [comment]
@jasperhuangg @rlinoz I wonder if y'all would know more about this one? Seems we've discussed something similar in the past

**Comment by sakluger** [comment]
No updates

**Comment by sakluger** [comment]
@Beamanator @jasperhuangg @rlinoz any ideas?

**Comment by rlinoz** [comment]
I think a lot of this has been updated since the issue was opened, this is the email that I got 

<img width="507" alt="Image" src="https://github.com/user-attachments/assets/d83ce5f4-0e2b-4300-a8a1-16ca0d52d6fa" />

One for the sender and one for the recipient, but basically the same email, what do we want to change here?

**Comment by sakluger** [comment]
Sounds like this may be fixed. @jamesdeanexpensify do you mind testing this out one more time to see if you can reproduce? If not, we can close.

**Comment by jamesdeanexpensify** [comment]
Cool! I only got one email so far, the correct one:

<img width="753" alt="Image" src="https://github.com/user-attachments/assets/77e3ae6b-7d55-43b4-bb45-fc8ff547fb2e" />

**Comment by cristipaval** [comment]
I'm reopening this one, it reproduced to me today
https://expensify.slack.com/archives/C049HHMV9SM/p1746435282250079

**Comment by jamesdeanexpensify** [comment]
Just tried another send to @sakluger to confirm this is still a bug.

**Comment by jamesdeanexpensify** [comment]
We should def get this fixed before F1, it makes us look like rookies!

**Comment by sakluger** [comment]
No updates.

**Comment by sakluger** [comment]
The PR is live! 🚀 

Looks like it's fixed! 🎉 

<img width="899" alt="Image" src="https://github.com/user-attachments/assets/ec4b43b6-7d2f-4cb0-83ad-3ed42db3a5ca" />
